### PR TITLE
Added abstract game class

### PR DESCRIPTION
**Pull request by schultsa**
Added Abstract Game class and implemented the Build deck method and remove method in each child class.

### CODE REVIEW

**Comment by schultsa** [comment]
Build failed due to not having Dennis's changes yet, the controller still tries to instantiate the abstract game class instead of the child classes.

File: src/main/java/models/Game.java [hunk 1]
```diff
@@ -116,41 +53,19 @@ public void customDeal(int c1, int c2, int c3, int c4) {
...
-    private boolean colHasCards(int colNumber) {
+    protected boolean colHasCards(int colNumber) {
         if(this.cols.get(colNumber).size()>0){
             return true;
         }
         return false;
     }
 
-    private Card getTopCard(int columnNumber) {
+    protected Card getTopCard(int columnNumber) {
```
Comment (charan96) [inline src/main/java/models/Game.java:68]: why is this protected? Is there a specific reason??

File: src/main/java/models/Card.java [hunk 2]
```diff
@@ -11,23 +11,41 @@
...
-    public final Suit suit;
+    public americanSuit amSuit;
+    public spanishSuit spSuit;
 
     @JsonCreator
-    public Card(@JsonProperty("value") int value, @JsonProperty("suit") Suit suit) {
+    public Card(@JsonProperty("value") int value, @JsonProperty("suit") americanSuit amSuit, @JsonProperty spanishSuit spSuit)
+    {
         this.value = value;
+<<<<<<< HEAD
```
Comment (dennislee15) [inline src/main/java/models/Card.java:21]: You should remove these before creating the pull request. I will remove everything between the HEAD and the "======" as well as the long string of chars and numbers.

File: src/main/java/models/Game.java (hunk 1 above)
Comment (schultsa) [inline src/main/java/models/Game.java:68]: I changed them to protected in order for the child game classes to be able to access them while still providing a level of security that Public does not offer.

File: src/main/java/models/Game.java [hunk 3]
```diff
@@ -22,71 +23,7 @@ public Game(){
...
-
-
-
-
-
-
-
-
-
+    public abstract void buildDeck();
```
Comment (charan96) [inline src/main/java/models/Game.java:26]: it's good that you made the buildDeck abstract, so that you can override this function in the child classes

**Pull request merged**